vertx/tests-vertx5/src/test/java/feign/vertx/RetryingTest.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright © 2012 The Feign Authors (feign@commonhaus.dev)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package feign.vertx;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
+import static com.github.tomakehurst.wiremock.stubbing.Scenario.STARTED;
+import static feign.vertx.tests.TestUtils.MAPPER;
+import static feign.vertx.tests.testcase.domain.Flavor.FLAVORS_JSON;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import feign.Logger;
+import feign.RetryableException;
+import feign.Retryer;
+import feign.VertxFeign;
+import feign.jackson.JacksonDecoder;
+import feign.slf4j.Slf4jLogger;
+import feign.vertx.tests.testcase.IcecreamServiceApi;
+import feign.vertx.tests.testcase.domain.Flavor;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.ext.web.client.WebClient;
+import io.vertx.junit5.VertxTestContext;
+import java.util.Arrays;
+import java.util.Collection;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+@DisplayName("When server ask client to retry")
+class RetryingTest extends AbstractFeignVertxTest {
```

Comment:
These look to be exactly the same

https://github.com/OpenFeign/feign/pull/2756/files#diff-618b10051163eb35607cd662bc192620411ccce4da0be7a99ee6dfb0b0ee4ad9L50

Why not move to test commons and just have a simplified version here?